src/AppBundle/Admin/AdministratorAdmin.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace AppBundle\Admin;
+
+use AppBundle\Entity\Administrator;
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\AdminBundle\Datagrid\DatagridMapper;
+use Sonata\AdminBundle\Form\FormMapper;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\PasswordType;
+use Symfony\Component\Form\Extension\Core\Type\RepeatedType;
+use Symfony\Component\Security\Core\Encoder\EncoderFactoryInterface;
+
+class AdministratorAdmin extends AbstractAdmin
+{
+    /**
+     * @var EncoderFactoryInterface
+     */
+    private $encoders;
+
+    /**
+     * @param Administrator $admin
+     */
+    public function prePersist($admin)
+    {
+        parent::prePersist($admin);
+
+        $admin->setPassword($this->encoders->getEncoder($admin)->encodePassword($admin->plainPassword, null));
```

Comment:
I'm not sure to understand what you mean, and how I could integrate this in Sonata. Could you give me more informations?